libsql-server/src/lib.rs

```diff
@@ -365,6 +365,7 @@ where
         let snapshot_callback = self.make_snapshot_callback();
         let auth = self.user_api_config.get_auth().map(Arc::new)?;
         let extensions = self.db_config.validate_extensions()?;
+        let mut primary_namespace_store: Option<NamespaceStore<PrimaryNamespaceMaker>> = None;
```

Comment:
I couldn't think of a better place than here to keep the NamespaceStore reference

Reply:
There is another way you can do that, which doesn't leak the primary implementation details to a wider scope: you put it in a future. You rename this to `namespace_store_shutdown_fut: Pin<Box<dyn Future<Output = Result<()>>>>`, and declare it as:
```rust
namespace_store_shutdown_fut = Box::pin(async move {
    namespace_store.shutdown().await
})
```

in both the replica and primary branch :)

The advantage of that is that if we decide  to extend the shutdown method for replica, everything is already in place to do that